docs/ref-manual/approx/array-hashes.md

```diff
@@ -0,0 +1,44 @@
+Array hashes
+============
+
```

Comment:
Somewhere up front we should clearly explain the approximation.  "The Prover does not model hashing directly.  Instead, it makes the following assumptions about `hash(x)`: - if `x != y` then `hash(x) != hash(y)`, etc.

We should also clearly explain the surprising thing the user might see.  A clear example of a rule that should pass and doesn't, or a rule that shouldn't pass and does, and an explanation of why.